### DISCUSSION TITLE
Recommendations for a graph consisting of thousands of nodes and edges

### DISCUSSION THREAD

**Discussion by kwitaszczyk** [terrastruct/d2#1809] [General]
I'm trying to generate a diagram with a graph consisting of around 27K nodes and 82K edges. The input file is around 4MB. It's a function call graph of the FreeBSD kernel. Unfortunately, the process has not finished after 42h.

Does anyone have any recommendations how to approach building a diagram of this scale? I guess one option would be to fragment it into smaller diagrams and manually join them. However, I wonder if I'm missing some configuration flags I could use to improve the performance of `d2`.

Command: `d2 kernel.full.callgraph.d2 kernel.full.callgraph.png`
d2 version: 0.6.3
Host: macOS 14.2.1, Macbook Pro, 2020, 2.3 GHz Quad-Core Intel Core i7, 32 GB DRAM

**Comment by alixander** [comment]
I've never tested D2 on that scale, and don't plan to support the use case.

I've heard good things about https://www.graphistry.com/ , maybe you could give them a try.

**Reply by alixander** [reply]
https://github.com/terrastruct/d2-docs/issues/186#issuecomment-1902760065

**Reply by kwitaszczyk** [reply]
Thanks for clarifying, and the recommendation.
